art/attacks/HCLU.py

```diff
@@ -0,0 +1,118 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+import abc
+import sys
+import copy
+import numpy as np
+from scipy.optimize import minimize
+from art.attacks.attack import Attack
+from art.classifiers.GPy import GPyGaussianProcessClassifier
+logger = logging.getLogger(__name__)
+
+
+class HCLU(Attack):
+    """
+    # Implementation of the HCLU formulation in https://arxiv.org/abs/1812.02606 by Grosse et al.  
+    """
+    attack_params = ['conf', 'unc_increase']
+
+    def __init__(self, classifier, conf=0.95, unc_increase=100.0, minVal=0.0, maxVal=1.0):
+        """
+        :param classifier: A trained model of type scikit decision tree.
+        :type classifier: :class:`.Classifier.GPyGaussianProcessClassifier
+        :param conf: Confidence that examples should have, if there were to be classified as 1.0 maximally
+        :type conf: :float: 
+        :param unc_increase: Value uncertainty is allowed to deviate, where 1.0 is original value
+        :type unc_increase: :float:
+        :param minVal: minimal value any feature can take
+        :type minVal: :float:
+        :param maxVal: maximal value any feature can take
+        :type maxVal: :float:
+        """
+        self.classifier = classifier
```

Comment:
Please call first `__init__` of the super class: `super(HCLU, self).__init__(classifier=classifier)`
This replaces `self.classifier = classifier` which can be removed